tests/functional/clients/standalone/package_list/test_alpine.py

```diff
@@ -0,0 +1,46 @@
+import sys
+from .fixtures import alpine
+from . import path_params
+import json
+import pytest
+
+
+class TestAlpinePaths:
+
+    @pytest.mark.parametrize('path', path_params(alpine.pkgfiles_all))
+    def test_pkgfiles_all(self, analyzed_data, path):
+        result = analyzed_data("py38")
+        pkgs = result['image']['imagedata']['analysis_report']['package_list']['pkgfiles.all']['base']
+        assert pkgs.get(path) == 'APKFILE'
+
+    @pytest.mark.parametrize('pkg,version', [pytest.param(pkg, version, id=pkg) for pkg, version in alpine.pkgs_all.items()])
+    def test_pkgs_all(self, analyzed_data, pkg, version):
+        result = analyzed_data("py38")
+        pkgs = result['image']['imagedata']['analysis_report']['package_list']['pkgs.all']['base']
+        assert pkgs.get(pkg) == version
+
+    @pytest.mark.parametrize('pkg,metadata', alpine.pkgs_allinfo.items())
+    def test_pkgs_allinfo(self, analyzed_data, pkg, metadata):
+        result = analyzed_data("py38")
+        pkgs = result['image']['imagedata']['analysis_report']['package_list']['pkgs.allinfo']['base']
+        loaded = dict(pkgs.get(pkg, {}))
+        
+        # a separate test exists just for the files attribute, leave this out of the assertion
+        loaded.pop('files')
+        metadata = dict(metadata)
+        metadata.pop('files')
+
+        assert loaded == metadata
+
+    @pytest.mark.parametrize('pkg,metadata', alpine.pkgs_allinfo.items())
+    def test_pkgs_allinfo_files(self, analyzed_data, pkg, metadata):
+        result = analyzed_data("py38")
+        pkgs = result['image']['imagedata']['analysis_report']['package_list']['pkgs.allinfo']['base']
+        loaded = pkgs.get(pkg, {})
+        assert set(loaded['files']) == set(metadata['files'])
```

Comment:
this implies that `loaded['files']` has duplicates that are different from `metadata['files']`, is my assumption correct? Or what is the benefit of using `set()` here?

Reply:
this is setup for when running the tests against syft later --syft has a different ordering of the files list than what engine outputs. By testing files explicitly with set instead of a nested comparison (which will check the order of items) we can assert the data is correct without going to too much effort.

I can change this to a sorting of both lists instead of set to ensure the test fails if there are duplicates.